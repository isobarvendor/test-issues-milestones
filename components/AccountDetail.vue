<template>
<div class="account-detail">
    <h3> Account Details </h3>
     <div class="registration-body">
    <div class="details">
        <div class="input-container">
            <v-text-field
            v-model="names"
            label=""
          ></v-text-field>
             <v-text-field
            v-model="email"
          ></v-text-field>
        </div>
     </div>
    <div class="profile-icon">
        <img src="/img/icons/pic.png"/>
    </div>
    </div>
    <div class="date-container">
        <p>Date of birth</p>
          <v-app>
              <v-row>
      <v-col cols="12" md="6" xs="12" sm="12">
              <v-menu
          ref="menu"
          v-model="menu"
          :close-on-content-click="false"
          :return-value.sync="date"
          transition="scale-transition"
          top
          min-width="290px"
          dark
        >
          <template v-slot:activator="{ on, attrs }">
            <v-text-field
                v-model="date"
              readonly
              v-bind="attrs"
              v-on="on"
             prepend-icon="mdi-calendar"
            ></v-text-field>
          </template>
          <v-date-picker v-model="date" no-title scrollable>
            <v-spacer></v-spacer>
            <v-btn text color="primary" @click="menu = false">Cancel</v-btn>
            <v-btn text color="primary" @click="$refs.menu.save(date)">OK</v-btn>
          </v-date-picker>
        </v-menu>
      </v-col>
              </v-row>
          </v-app>
        </div>
        <div class="button-container">
            <a  class="button">Save</a>
        </div>

</div>
</template>

<script>
export default {
  name: "AccountDetail",
  components: {
  },
  data() {
    return {
        names: "Jane",
        email: "janet@gmail.com",
        date: new Date().toISOString().substr(0, 10),
        menu: false
    };
  },
  props: {
    data: null,
  },
  computed:{
  },
  methods:{
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.account-detail{
    margin-top: 40px;
    .theme--light.v-application{
        background: none !important;
    }
    .v-application--wrap{
        min-height:auto !important;
    }
    .theme--light.v-input input{
        color: white;
    }
    .theme--light.v-text-field > .v-input__control > .v-input__slot:before{
        border-color: #585858;
    }
    .mdi:before, .mdi-set{
        color: white;
    }
    h3{
        font-size: 25px;
        text-transform: none;
    }
    .registration-body{
        display: flex;
        justify-content: space-between;
        align-items: center;
        
        .v-text-field__slot{
            border-bottom: 1px solid #585858;
        }
        .theme--light.v-label{
            color: #585858;
        }
        .theme--light.v-input input{
            color: #ffffff;
        }
        .details{
            margin-top: 20px;
            width: 50%;
            
        }
        .profile-icon{
        
            width: 235px;
            margin: auto;
                
        }
        .input-container{
            margin-top: 25px;
        }
        .date-container{
            display: flex;
            align-items: center;
            .radio-label{
                font-size: 14px;
                margin-left: 10px;
                line-height: 24px;
            }
            .v-input--selection-controls__input .v-icon{
                color: #585858;
            }
        }
        .button-container{
            margin-top: 40px;
            a.button{
                width: 230px;
                display: block;
            }
        }
        
    }
    @media only screen and (max-width: 1199px) {
        .registration-body{
            flex-direction: column-reverse;
            .details{
                width: 100%;
            }
            .profile-icon{
                margin-top: 35px;
            }
            a.button{
                margin:auto;
            }
        } 
        h3{
            text-align: center;
            font-size: 22px;
        }
        .button-container a.button{
            margin: auto;
        }   
    }  
}
</style>
