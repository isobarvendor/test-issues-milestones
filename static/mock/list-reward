[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
