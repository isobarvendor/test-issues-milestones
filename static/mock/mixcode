{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
