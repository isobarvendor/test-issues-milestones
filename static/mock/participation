{"id":16,"questionId":1,"answerId":3,"participationId":17}
