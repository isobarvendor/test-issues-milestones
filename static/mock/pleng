{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p>3 months subscription to Pleng Pro</p>",
      "shortDescription": "<p>ឈ្នះថ្លៃសេវា Pleng Pro Subscription រយៈពេល 3 ខែ</p>",
      "priority": 1,
      "prizeId": "ks8o4ixh",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8o4ixh/CAM prizes — Pleng Pro@2x.png",
      "name": "3 months subscription to Pleng Pro ",
      "redemptionLink": "https://pleng.onelink.me/pp89?pid=ddlink&af_web_dp=https%3A%2F%2Fwww.smart.com.kh%2Fkm%2Fpleng%2F&af_dp=plengmusic%3A%2F% 2F&af_partner_name=cc",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះ Pleng Pro Subscription រយៈពេល ៣ ខែ! រីករាយនូវការស្តាប់ចម្រៀងឥតដែនកំណត់។ សូមដោនឡូតកម្មវិធី Pleng ពី Smart ដើម្បីបញ្ចូលលេខកូដ និងដាក់ដំណើរការ Smart Pleng Pro។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
      
    }
  }
}