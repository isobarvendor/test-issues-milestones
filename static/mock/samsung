{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p></p>",
      "shortDescription": "<p>ទូរស័ព្ទសាមសុង Galaxy Note20 Ultra</p>",
      "priority": 1,
      "prizeId": "ks8ns5hv",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8ns5hv/1.CAM prize-N20 Ultra@2x.png",
      "totalAmount": 20,
      "minAge": 13,
      "name": "Samsung Galaxy Note20 Ultra",
      "redemptionLink": "",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះទូរស័ព្ទសាមសុង Galaxy Note20 Ultra! សូមទាក់ទងទៅលេខ 1-800-20-2222 (មិនគិតថ្លៃសេវាទូរស័ព្ទ ២៤/៧) ដើម្បីដឹងពីរបៀបប្តូរយករង្វាន់។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ</p>",
      "voucherCode": "CNXFUUAKAY"
    }
  }
}