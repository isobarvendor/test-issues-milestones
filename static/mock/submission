{
            "burnResult": [
                {
                    "pincode": "0J6TC2VP5",
                    "programId": "453269",
                    "lotId": "445580",
                    "burned": true
                }
            ],
            "participationId": "1ylazdy9kkxmpa53",
            "instantWinResult": {
                "winner": true,
              "redeemedPrize": {
                    "prizeId": "kkqhy9so",
                    "voucherCode": "Voucher891",
                    "status": "claimed",
                    "expiryDate": 1620018590000,
                    "name": "JOOX VIP PASS",
                    "description":"Conglratulations",
                    "shortDescription": "http://google.com",
                    "redeemDescription": "http://abcd",
                    "imgUrl": "/img/landing/week 1 prize.png",
                    "barcodeType": 1,
                    "emailSent": false,
                    "emailMessage": "Email could not be sent"
                }
            }
        }
