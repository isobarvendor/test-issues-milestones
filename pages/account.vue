<template>
  <div v-if="dataStatus.status == 200" id="main" class="account">
    <div class="wrapper">
      <div class="container">
        <AccountDetail :data="data.exclusivePrizes"/>
      </div>

    </div>
  </div>
  <div v-else-if="dataStatus.status >= 500">
    Status: {{dataStatus.status}}
    <br />
    {{dataStatus.message}}
  </div>
</template>

<script>
import AccountDetail from '@/components/AccountDetail'
export default {
  data(){
    return{
      dataStatus:{},
      data:null,
    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],
      css: []
    };
  },
  created() {
    this.fetchData();
  },
  methods: {
    async fetchData() {
      let result = await this.$axios.get(
        "https://ayo.aircovery.com/cms-api/campaigns/1"
      );
      this.dataStatus = { status: result.status, message: result.statusText };
      this.data = result.data;
    },
  },
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    @media only screen and (max-width: 767px) {
      .account{
      }
    }
  }
</style>