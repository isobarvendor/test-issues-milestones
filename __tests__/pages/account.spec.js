import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'

import account from '@/pages/account.vue';

import AccountDetail from '@/components/AccountDetail'


import {mockCMSConfig,mockCMSContent} from "../../__mock__/mock";





let localVue, store,router;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    router = new VueRouter();
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        token:"",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      }
    });
  });

  const factory = () => {
    return mount(account, {

      localVue, store,router
    });
  };



describe('account', () => {


  test('renders a accountDetail', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(AccountDetail);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
