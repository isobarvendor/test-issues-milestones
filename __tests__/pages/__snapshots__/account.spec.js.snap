// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account renders properly 1`] = `
<div id="main" class="account">
  <div>
    <div class="wrapper">
      <div class="container">
        <div class="account-detail">
          <h3> accounts</h3>
          <div class="registration-body">
            <div class="details">
              <div class="input-container">
                <div class="v-input v-input--is-label-active v-input--is-dirty v-input--is-disabled theme--light v-text-field">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <div class="v-text-field__slot"><input disabled="disabled" id="input-31" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="v-input v-input--is-label-active v-input--is-dirty v-input--is-disabled theme--light v-text-field">
                  <div class="v-input__control">
                    <div class="v-input__slot">
                      <div class="v-text-field__slot"><input disabled="disabled" id="input-34" type="text"></div>
                    </div>
                    <div class="v-text-field__details">
                      <div class="v-messages theme--light">
                        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="profile-icon"><img src="tes"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
